beta-src/src/utils/map/drawMoveOrders.ts

```diff
@@ -1,32 +1,82 @@
 import TerritoryMap from "../../data/map/variants/classic/TerritoryMap";
 import ArrowColor from "../../enums/ArrowColor";
 import ArrowType from "../../enums/ArrowType";
+import { GameState } from "../../state/interfaces/GameState";
 import OrdersMeta from "../../state/interfaces/SavedOrders";
 import drawArrow from "./drawArrow";
 
-export default function drawMoveOrders(data, ordersMeta: OrdersMeta): void {
+export default function drawMoveOrders(
+  data: GameState["data"]["data"],
+  maps: GameState["maps"],
+  ordersMeta: OrdersMeta,
+  board: GameState["board"],
+): void {
   const { currentOrders, territories, units } = data;
   const ordersMetaEntries = Object.entries(ordersMeta);
   if (ordersMetaEntries.length && units && territories) {
     ordersMetaEntries
       .filter(([, { update }]) => update?.type === "Move")
       .forEach(([orderID, { update }]) => {
-        const originalOrder = currentOrders.find(({ id }) => id === orderID);
-        if (originalOrder && update && update.toTerrID) {
-          const { id, unitID } = originalOrder;
-          const unitData = units[unitID];
-          const onTerrDetails = territories[unitData.terrID];
-          const toTerrDetails = territories[update.toTerrID];
-          const fromTerr = TerritoryMap[onTerrDetails.name].territory;
-          const toTerr = TerritoryMap[toTerrDetails.name].territory;
-          drawArrow(
-            id,
-            ArrowType.MOVE,
-            ArrowColor.MOVE,
-            "territory",
-            toTerr,
-            fromTerr,
-          );
+        const originalOrder = currentOrders?.find(({ id }) => id === orderID);
+        if (originalOrder && update) {
+          const { convoyPath, toTerrID, viaConvoy } = update;
+          if (toTerrID) {
+            const { id, unitID } = originalOrder;
+            const unitData = units[unitID];
+            const onTerrDetails = territories[unitData.terrID];
+            const toTerrDetails = territories[toTerrID];
+            const fromTerr = TerritoryMap[onTerrDetails.name].territory;
+            const toTerr = TerritoryMap[toTerrDetails.name].territory;
+            drawArrow(
+              id,
+              ArrowType.MOVE,
+              ArrowColor.MOVE,
+              "territory",
+              toTerr,
+              fromTerr,
+            );
+            if (viaConvoy === "Yes" && board) {
+              // implied convoy arrow
+              const unitD = board.findUnitByID(unitID);
```

Comment:
seems a little confusing unitD and having unitID. Can we change unitD to something else. Is this the unit that will support the convoy?

unitIDConvoy maybe instead of unitD